### ISSUE TITLE
Fix n_alive component counting

### ISSUE THREAD

**Issue by danbraunai** [goodfire-ai/spd#143]
Lucius has noticed that the n_alive_0.1 metric always looks like garbage. E.g. in this tms_40-10-id [run](https://wandb.ai/goodfire/spd/runs/dsn94977), the n_alive_0.1 value jumps around at about 40 for the hidden_layers. We're expecting it to take a value of 10, because we just need 10 dense subcomponents to represent this identity transformation.

Recall that the alive components are measured by "If the component fires over the last 4M samples (batch size 4k), then consider it alive".

It appears that on some runs there are just some components that fire occasionally with small-ish causal importance values, though sometimes up to 0.8 (if I set a threshold of 0.8 then the above experiment usually does converge to 10 alive components, as we'd expect).

Other runs don't seem to have this property, and converge much sooner (e.g. like this [run](https://wandb.ai/goodfire/spd/runs/d1jxxmhr) which uses the inserted identity layers from [PR141](https://github.com/goodfire-ai/spd/pull/141)).

Adding weight decay didn't seem to improve things on my very rough test.

Note that another reason why we might see inflated n_alive_components is if a dead component effectively "swaps" with an alive component during training. From a very preliminary test I don't think this is happening: I stopped updating gradients after a bunch of steps and noticed similar n_alive values.

Temporary Solution
Setting the ci_alive_threshold to something like 0.3 will give you close to optimal scores on tms_40-10-id. It's probably OK to use a value like this for LM experiments. But we should at some point try and get this cleaner in the toy models (and test on other toy models, which I'm yet to do.

It might also be useful to track multiple values for this metric in wandb, e.g. 0, 0.1, 0.3, 0.8.

Note that we need this to be able to tell if our [inserted identity matrices](https://github.com/goodfire-ai/spd/pull/141) are being decomposed correctly

**Comment by danbraunai** [comment]
Not as important now that we have the CIMeanPerComponent stat. But it would still be great if this could be solved.

**Comment by oli-clive-griffin** [comment]
to be clear, are you claiming the logic is wrong? or just that the training dynamics don't work nicely with the threshold?
